src/hooks/useDebounceState/useDebounceState.types.ts

```diff
@@ -0,0 +1,10 @@
+export type UseDebounceStateProps = {
+  onDebounceChange: (value: number | undefined) => void;
+  initialValue: number | undefined;
+  debounceTime: number;
+};
+
+export type UseDebounceStateReturnedValue = [
+  number | undefined,
+  React.Dispatch<React.SetStateAction<number | undefined>>,
```

Comment:
Tu i powyżej - to nie musi być `number`, prawda? Dobrze by było z tego zrobić generyka, żeby w przypadku `stringa` też działał dobrze